pkg/controller/velero/velero.go

```diff
@@ -294,6 +352,35 @@ func veleroDeployment(namespace string, veleroImage string) *appsv1.Deployment {
 	return deployment
 }
 
+func veleroService(namespace string) *corev1.Service {
```

Comment:
So I'd like to future-proof this function a bit with a bit more autogeneration and less hard coding of values.

My suggestion:
we create a function that is `func createServiceFromDeployment(d *appsv1.Deployment) *corev1.Service`

Then based on the deployment, we can generate most of the details:
- Service name := `d.ObjectMeta.Name` + "-metrics"
- Namespace should match the deployment
- Loop through all containers and all the ports on the containers, and anyone that has "metrics" in the name should be added to the ServicePorts
- Selector should match the deployment.Spec.Template labelset

Reply:
:+1: I like it.